woocommerce/woocommerce-ios
# [Mobile Payments] Fix hardcoded date in receipts

Closes #4131 

⚠️ This PR is against the feature branch implementing support for Card Present Payments, not against develop ⚠️
⚠️ Testing this PR requires a hardware reader and uploading a pre-release version of WCPay to a test site⚠️

Oopsies. This one is on me. While working on other things (#4130) I noticed that all the receipts had March 23 as date. Why? Go figure. I guess that's when I wrote that part of the code.

Anyway, this one fixes that one. 

## Changes
* Add the PaymentIntent creation date (which I think is safe to assume that corresponds to the date the payment was collected) to the struct with the metadata needed to render the receipt
* Touch the code that renders the receipts to use that date

## How to test
**TL:DR; Collect a payment and generate a receipt. Notice the date in the receipt, it should match your date, and be formatted in in your locale.**

Testing requires:

1. Setting up a store for Card Present Payment Testing (also P91TBi-4BH-p2 for more specific instructions)
2. The pre-release version of WCPay that implements the endpoint that captures the payment id (see p91TBi-54R#comment-4441-p2 for a downloadable zip)
3. An external hardware reader.

* Checkout the branch, run bundle exec pod install, just for kicks and giggles.
* Build and run on a device.
* Switch an external card reader on. The only card reader supported by now is the BBPOS Chipper 2X BT.
* Log in to an account that matches a store that has been configured for testing.
* Navigate to Settings in the app. (Tap the gear button in the top right)
* In the "Store Settings" section, tap "Manage Card Reader".
* Make sure the card reader is on and the blue light is blinking (we don't do any error handling yet, so it's important that things are setup for the happy path)
* Tap "Connect card reader"
* Wait until the app suggests one reader to connect to.
* Tap "Connect to reader"
* Make sure the reader light turns solid blue.
* Navigate to an order eligible for Card Present Payments (it has to be a store enrolled to WCPay, and the order needs to have as payment method associated Cash on Delivery or none).
* Tap the "Collect Payment" button.
* Tap the card to the reader when requested.
* If everything goes well, the card should be charged, and the payment should be completed.
* Tap the "Email Receipt" button or "Print Receipt"
* If emailing, the standard iOS email composing sheet is presented. Notice the date
* When printing, the date should also be part of the receipt.


Update release notes:
